packages/devextreme/js/__internal/ui/m_switch.ts

```diff
@@ -309,67 +320,69 @@ class Switch extends Editor<Properties> {
-    this._renderPosition(this.option('value'), e.event.offset);
+  _swipeUpdateHandler(e: SwipeUpdateEvent): void {
+    const { value } = this.option();
+    this._renderPosition(Boolean(value), e.event.offset);
   }
 
-  _swipeEndHandler(e): void {
-    const that = this;
+  _swipeEndHandler(e: SwipeEndEvent): void {
+    const { value } = this.option();
+
     const offsetDirection = this._offsetDirection();
-    const toInnerConfig = {};
-    const toHandleConfig = {};
-
-    const innerOffset = this._getInnerOffset(that.option('value'), e.event.targetOffset);
-    const handleOffset = this._getHandleOffset(that.option('value'), e.event.targetOffset);
-    // @ts-expect-error
-    toInnerConfig.transform = ` translateX(${innerOffset})`;
-    // @ts-expect-error
-    toHandleConfig.transform = ` translateX(${handleOffset})`;
-    // @ts-expect-error ts-error
-    fx.animate(this._$handle, {
+
+    const innerOffset = this._getInnerOffset(Boolean(value), e.event.targetOffset);
+    const handleOffset = this._getHandleOffset(Boolean(value), e.event.targetOffset);
+
+    const toInnerConfig = { transform: ` translateX(${innerOffset})` };
+    const toHandleConfig = { transform: ` translateX(${handleOffset})` };
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    fx.animate(this._$handle.get(0), {
+      // @ts-expect-error AnimationState type should be extended
       to: toHandleConfig,
       duration: SWITCH_ANIMATION_DURATION,
     });
-    // @ts-expect-error ts-error
-    fx.animate(this._$switchInner, {
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    fx.animate(this._$switchInner.get(0), {
+      // @ts-expect-error AnimationState type should be extended
       to: toInnerConfig,
       duration: SWITCH_ANIMATION_DURATION,
-      complete() {
-        that._swiping = false;
-        // @ts-expect-error ts-error
-        const pos = that.option('value') + offsetDirection * e.event.targetOffset;
-        that._saveValueChangeEvent(e.event);
-        that.option('value', Boolean(pos));
-        that._feedbackDeferred?.resolve();
-        that._toggleActiveState(that.$element(), false);
+      complete: () => {
+        this._swiping = false;
+        const pos = Number(value) + offsetDirection * e.event.targetOffset;
+        this._saveValueChangeEvent(e.event);
+        this.option({ value: Boolean(pos) });
+        this._feedbackDeferred?.resolve();
+        this._toggleActiveState(this.$element(), false);
       },
     });
   }
 
   _renderValue(): void {
     this._validateValue();
 
-    const { value } = this.option();
-    this._renderPosition(value, 0);
+    const { value, switchedOnText, switchedOffText } = this.option();
+
+    this._renderPosition(Boolean(value), 0);
 
     this.$element().toggleClass(SWITCH_ON_VALUE_CLASS, value);
-    // @ts-expect-error ts-error
-    this._getSubmitElement().val(value);
+
+    this._getSubmitElement().val(String(value));
```

Comment:
I'm not sure the type coercion is safe here - in this case, it would change the behavior.
```
val(null) → ''
val(undefined) → ''
val(String(null)) → 'null'
val(String(undefined)) → 'undefined'
```
and others cases. Kindly validate it

Reply:
Boolean value will convert to string type as well